utilities/generate_messages_catalog.py

```diff
@@ -131,96 +134,169 @@ def _is_callable(item):
-                try:
-                    # imported function could be by id instead of attr
-                    handler = FUNC_HANDLER.get(
-                            item.func.attr, lambda x:  ("", 0)
-                    )
-                    if isinstance(handler, tuple):
-                        level, args_offset = handler
-                    else:
-                        level, args_offset = handler(item)
-
-                    msgCodeArg = item.args[0 + args_offset]  # str or tuple
-                    if isinstance(msgCodeArg,ast.Str):
-                        msgCodes = (msgCodeArg.s,)
+    ref_module_name = os.path.basename(python_module)
+    with open(python_module, encoding="utf-8") as module_file:
+        tree = ast.parse(module_file.read(), filename=python_module)
+        callables = filter(_is_callable, ast.walk(tree))
+        for item in callables:
+            try:
+                # imported function could be by id instead of attr
+                handler = FUNC_HANDLER.get(
+                    item.func.attr, lambda x: ("", 0)
+                )
+                level, args_offset = handler
+
+                msgCodeArg = item.args[0 + args_offset]  # str or tuple
+                if isinstance(msgCodeArg, ast.Str):
+                    msgCodes = (msgCodeArg.s,)
+                else:
+                    if any(isinstance(element, (ast.Call, ast.Name))
+                           for element in ast.walk(msgCodeArg)):
+                        msgCodes = ("(dynamic)",)
                     else:
-                        if any(isinstance(element, (ast.Call, ast.Name))
-                               for element in ast.walk(msgCodeArg)):
-                            msgCodes = ("(dynamic)",)
+                        msgCodes = [
+                            element.s for element in ast.walk(msgCodeArg)
+                            if isinstance(element, ast.Str)
+                        ]
+                msgArg = item.args[1 + args_offset]
+                if isinstance(msgArg, ast.Str):
+                    msg = msgArg.s
+                elif ((isinstance(msgArg, ast.Call) and
+                       getattr(msgArg.func, "id", '') == '_')):
+                    msg = msgArg.args[0].s
+                elif ((any(isinstance(element, (ast.Call,ast.Name))
+                       for element in ast.walk(msgArg)))):
+                    msg = "(dynamic)"
+                else:
+                    continue # not sure what to report
+                keywords = []
+                for keyword in item.keywords:
+                    if keyword.arg == 'modelObject':
+                        pass
+                    elif keyword.arg == 'messageCodes':
+                        msgCodeArg = keyword.value
+                        if ((any(isinstance(element, (ast.Call, ast.Name))
+                             for element in ast.walk(msgCodeArg)))):
+                            pass  # dynamic
```

Comment:
Nope, as intended. Because `if (` is exactly 4 characters, the same as a standard python indent, if you indent with only 4 characters multiline if statements, you get a flake8 error for ambitious indentation. To avoid this, the styleguide was amended to recommend a double paren forcing the alignment to five spaces, making for a clear distinction from the multiline if and the new context being opened. Reference [here](https://github.com/Workiva/styleguide/blob/master/PYTHON.rst#multi-line-ifelifwith-statements)